core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -0,0 +1,183 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Primary controller for the collection editor page.
+ *
+ * @author mgowano@google.com (Abraham Mgowano)
+ */
+
+// TODO(bhenning): These constants should be provided by the backend.
+oppia.constant(
+  'COLLECTION_DATA_URL_TEMPLATE', '/collection_handler/data/<collection_id>');
+oppia.constant(
+  'WRITABLE_COLLECTION_DATA_URL_TEMPLATE',
+  '/collection_editor_handler/data/<collection_id>');
+
+oppia.controller('CollectionEditor', ['$scope',
+  'WritableCollectionBackendApiService', 'SkillListObjectFactory',
+  'warningsData', function(
+    $scope, WritableCollectionBackendApiService, SkillListObjectFactory,
+    warningsData) {
+    $scope.collection = null;
+    $scope.collectionId = GLOBALS.collectionId;
+    $scope.collectionSkillList = SkillListObjectFactory.create([]);
+
+    // Get the id of the collection to be loaded
+    var pathnameArray = window.location.pathname.split('/');
+    for (var i = 0; i < pathnameArray.length; i++) {
+      if (pathnameArray[i] === 'create') {
+        $scope.collectionId = pathnameArray[i + 1];
+        break;
+      }
+    }
+
+    // Load the collection to be edited.
+    WritableCollectionBackendApiService.fetchWritableCollection(
+      $scope.collectionId).then(
+        function(collection) {
+          $scope.collection = collection;
+          $scope.collectionSkillList.setSkills(collection.skills);
+        }, function(error) {
+          warningsData.addWarning(
+            error || 'There was an error loading the collection.');
+        });
+
+    var _getCollectionNodeByExplorationId = function(explorationId) {
+      for (var i = 0; i < $scope.collection.nodes.length; i++) {
+        var collectionNode = $scope.collection.nodes[i];
+        if (collectionNode.exploration_id == explorationId) {
+          return collectionNode;
+        }
+      }
+      return null;
+    };
+
+    // To be used after mutating the prerequisite and/or acquired skill lists.
+    var _resetSkillList = function() {
+      // TODO(bhenning): This can be made far more optimal once individual
+      // prerequisite and acquired skill lists are also stored in the skill list
```

Comment:
I'm not sure this is a good idea? The individual lists seem to be to be more correlated with the nodes than with the global skill list.